client/context/load.go

```diff
@@ -0,0 +1,123 @@
+package context
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+
+	"github.com/docker/docker/pkg/contextstore"
+	"github.com/docker/go-connections/tlsconfig"
+	"github.com/pkg/errors"
+)
+
+// Context is a typed wrapper around a context-store context
+type Context struct {
+	Name          string
+	Host          string
+	SkipTLSVerify bool
+	APIVersion    string
+}
+
+// TLSData holds ca/cert/key raw data
+type TLSData struct {
+	CA   []byte
+	Key  []byte
+	Cert []byte
+}
+
+// LoadTLSData loads TLS materials for the context
+func (c *Context) LoadTLSData(s contextstore.Store) (*TLSData, error) {
+	tlsFiles, err := s.ListContextTLSFiles(c.Name)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to retrieve context tls info")
+	}
+	if epTLSFiles, ok := tlsFiles[dockerEndpointKey]; ok {
+		var tlsData TLSData
+		for _, f := range epTLSFiles {
+			data, err := s.GetContextTLSData(c.Name, dockerEndpointKey, f)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to retrieve context tls info")
+			}
+			switch f {
+			case caKey:
+				tlsData.CA = data
+			case certKey:
+				tlsData.Cert = data
+			case keyKey:
+				tlsData.Key = data
+			}
```

Comment:
`default: «warn about unknown key»`? Perhaps this is deliberate silently ignored (in which case perhaps a comment, maybe inside a dummy `default` case).